File: done/ccc03j2.py
from math import sqrt

d = int(input())
while(d):
    for i in range(int(sqrt(d)), 0, -1):
        if d % i == 0:
            print("Minimum perimeter is", 2 * (i + d // i), "with dimensions", i, "x", d // i)
            break
    d = int(input())

File: done/ccc03j2.c
#include <stdio.h>
#include <math.h>

int main() {
	int c, x;
	while (1) {
		scanf("%d", &c);
		if (!c)
			break;
		x = sqrt(c);
		do {
			if (c % x == 0) 
				break;
		}
		while (x-- > 0);
		
		printf("Minimum perimeter is %d with dimensions %d x %d\n", 2 * x + 2 * (c / x), x, c / x);
	}
	return 0;
}